<h2 style="text-align: center;">
Exploring Export Financing Trends: A Data-Driven Analysis of FY25 Q3 Records
</h2>

***Project Overview***

**Domain**: Banking and Finance

**Objective**: This project analyzes the FY25 Q3 export financing dataset from the U.S. Export-Import Bank (Data.gov).The goal is to clean, explore, and visualize the data to uncover key insights such as approval trends, top countries and products financed, program comparisons, support for diverse businesses, seasonal patterns, and financial risk exposure.By turning raw records into structured analysis, the project highlights how finance and banking analytics can guide decision-making in international trade.

**Problem Definition**: International trade financing involves complex approval processes, diverse business categories, and varying financial risks.  
Decision-makers often struggle to identify clear patterns in approvals, country-level financing, product trends, and support for small or minority-owned businesses. This project addresses the problem of turning raw government financing records into actionable insights. By applying data cleaning, exploratory analysis, and visualization techniques, the project aims to:
- Reveal approval trends and seasonal patterns  
- Compare financing across countries, products, and programs  
- Assess support for diverse businesses (small, women-owned, minority-owned)  
- Highlight financial risk exposure through outstanding loan analysis  
The outcome provides a structured view of export financing trends, enabling better understanding of how government support impacts international trade and business growth.

**Step1-Data Loading and Initial Overview**

**Goal**: Import the dataset using Pandas and understand the characteristics of the data

**Data Source**: https://img.exim.gov/s3fs-public/dataset/vbhv-d8am/Data.Gov_-_FY25_Q3.csv

**Dataset Description**:The FY25 Q3 export financing dataset, sourced from the U.S. Export-Import Bank via Data.gov, contains thousands of records and 33 attributes covering loan approvals, disbursements, outstanding exposures, interest rates, and program details. It spans Fiscal Year 2025, Quarter 3 (Oct 2024–Sep 2025) and includes categorical fields such as country, program type, policy type, and decision status, along with numerical fields like approved amounts and business support for small, women-owned, and minority-owned enterprises. With its mix of financial, categorical, and temporal data, the dataset provides a comprehensive view of government-backed export financing, enabling analysis of approval trends, country and product distributions, program comparisons, business support levels, and financial risk exposure.


**1.1 Importing Libraries**

We begin by importing the necessary Python libraries:
- **pandas** for data manipulation and analysis  
- **numpy** for numerical operations  
We also set display options to show all columns and format decimals to two places for cleaner outputs.

**1.2 Loading the Dataset**

The dataset is loaded directly from the official U.S. Export-Import Bank (Data.gov) source using `pd.read_csv()`.  

**1.3 Inspecting the Dataset**

After loading, we perform an initial inspection to understand the structure and quality of the data:
- **Shape:** Number of rows and columns in the dataset  
- **Column Names:** List of all attributes available for analysis  
- **Data Types:** Identify whether each column is numerical, categorical, or date  
- **Sample Records:** Display the first and last 10 rows  to preview actual values  
- **Info & Summary:** Use .info() and .describe() to check data completeness, types, and basic statistics 
- **Summary of Findings:** This step helps to find an overview of dataset quality and key business insights. It will identify potential issues such as missing values, mixed data types, or redundant columns, which will be addressed during data pre-processing.


***1.1 Importing Libraries***

In [1]:
# Importing Libraries
import pandas as pd
import numpy as np

# Display settings
pd.set_option('display.max_columns', None)
pd.set_option('display.float_format', '{:.2f}'.format)

# Confirmation message
print("Libraries loaded successfully")

Libraries loaded successfully


We import essential Python libraries, pandas and numpy 

We also set display options to show all columns and format. Pandas hides some columns if there are too many, showing ... in the middle. With this option all columns are displayed in outputs. We can Confirm the Libraries were loaded successfully by the confirmation message.

***1.2 Loading the Dataset***

In [2]:
#  Loading the Dataset
url = "https://img.exim.gov/s3fs-public/dataset/vbhv-d8am/Data.Gov_-_FY25_Q3.csv"
df = pd.read_csv(url, low_memory=False)

print("Dataset loaded successfully!")
print("Shape (rows, columns):", df.shape)

# Export the dataset to a CSV file
df.to_csv("exported_dataset.csv", index=False)

print("Dataset exported successfully as exported_dataset.csv")

Dataset loaded successfully!
Shape (rows, columns): (51753, 34)
Dataset exported successfully as exported_dataset.csv


Dataset successfully loaded and exported with 51,753 rows and 34 columns.

***1.3 Inspecting the Dataset***

**1.3.1 Column Names**

In [3]:
# 1.3.2 Column names
list(df.columns)

['Fiscal Year',
 'Unique Identifier',
 'Deal Number',
 'Decision',
 'Decision Date',
 'Effective Date',
 'Expiration Date',
 'Brokered ',
 'Deal Cancelled',
 'Country',
 'Program',
 'Policy Type',
 'Decision Authority',
 'Primary Export Product NAICS/SIC code',
 'Product Description',
 'Term',
 'Primary Applicant',
 'Primary Lender',
 'Primary Exporter',
 'Primary Exporter City',
 'Primary Exporter State Code',
 'Primary Exporter State Name',
 'Primary Borrower',
 'Primary Source of Repayment (PSOR)',
 'Working Capital Delegated Authority',
 'Approved/Declined Amount',
 'Disbursed/Shipped Amount',
 'Undisbursed Exposure Amount',
 'Outstanding Exposure Amount',
 'Small Business Authorized Amount',
 'Woman Owned Authorized Amount',
 'Minority Owned Authorized Amount',
 'Loan Interest Rate',
 'Multiyear Working Capital Extension ']

The columns include both deal details and business information, making the dataset useful for a complete analysis of financing trends.

**Dataset Columns and Their Meaning**:

| Column Name                                | Meaning                                                                 |
|--------------------------------------------|-------------------------------------------------------------------------|
| Fiscal Year                                | The fiscal year in which the deal was recorded                          |
|--------------------------------------------|-------------------------------------------------------------------------|
| Unique Identifier                          | A unique code assigned to each record                                   |
|--------------------------------------------|-------------------------------------------------------------------------|
| Deal Number                                | Internal reference number for the financing deal                        |
|--------------------------------------------|-------------------------------------------------------------------------|
| Decision                                   | Status of the deal (Approved / Declined)                                |
|--------------------------------------------|-------------------------------------------------------------------------|
| Decision Date                              | Date when the financing decision was made                               |
|--------------------------------------------|-------------------------------------------------------------------------|
| Effective Date                             | Date when the financing agreement became effective                      |
|--------------------------------------------|-------------------------------------------------------------------------|
| Expiration Date                            | Date when the financing agreement expires                               |
|--------------------------------------------|-------------------------------------------------------------------------|
| Brokered                                   | Indicates if the deal was brokered through a third party                |
|--------------------------------------------|-------------------------------------------------------------------------|
| Deal Cancelled                             | Shows if the deal was later cancelled                                   |
|--------------------------------------------|-------------------------------------------------------------------------|
| Country                                    | Destination country for the export financing                            |
|--------------------------------------------|-------------------------------------------------------------------------|
| Program                                    | Financing program under which the deal was processed                    |
|--------------------------------------------|-------------------------------------------------------------------------|
| Policy Type                                | Type of policy applied to the financing deal                            |
|--------------------------------------------|-------------------------------------------------------------------------|
| Decision Authority                         | Authority or official who approved/declined the deal                    |
|--------------------------------------------|-------------------------------------------------------------------------|
| Primary Export Product NAICS/SIC code      | Industry classification code of the exported product                    |
|--------------------------------------------|-------------------------------------------------------------------------|
| Product Description                        | Description of the exported product/service                             |
|--------------------------------------------|-------------------------------------------------------------------------|
| Term                                       | Duration or repayment term of the financing                             |
|--------------------------------------------|-------------------------------------------------------------------------|
| Primary Applicant                          | Main applicant requesting financing                                     |
|--------------------------------------------|-------------------------------------------------------------------------|
| Primary Lender                             | Financial institution or lender involved                                |
|--------------------------------------------|-------------------------------------------------------------------------|
| Primary Exporter                           | Exporter company associated with the deal                               |
|--------------------------------------------|-------------------------------------------------------------------------|
| Primary Exporter City                      | City where the exporter is located                                      |
|--------------------------------------------|-------------------------------------------------------------------------|
| Primary Exporter State Code                | State code of the exporter’s location                                   |
|--------------------------------------------|-------------------------------------------------------------------------|
| Primary Exporter State Name                | Full state name of the exporter’s location                              |
|--------------------------------------------|-------------------------------------------------------------------------|
| Primary Borrower                           | Borrower company receiving financing                                    |
|--------------------------------------------|-------------------------------------------------------------------------|
| Primary Source of Repayment (PSOR)         | Main source of repayment for the loan                                   |
|--------------------------------------------|-------------------------------------------------------------------------|
| Working Capital Delegated Authority        | Indicates if working capital authority was delegated                    |
|--------------------------------------------|-------------------------------------------------------------------------|
| Approved/Declined Amount                   | Amount approved or declined for financing                               |
|--------------------------------------------|-------------------------------------------------------------------------|
| Disbursed/Shipped Amount                   | Amount actually disbursed or shipped                                    |
|--------------------------------------------|-------------------------------------------------------------------------|
| Undisbursed Exposure Amount                | Amount authorized but not yet disbursed                                 |
|--------------------------------------------|-------------------------------------------------------------------------|
| Outstanding Exposure Amount                | Current outstanding balance of financing                                |
|--------------------------------------------|-------------------------------------------------------------------------|
| Small Business Authorized Amount           | Financing amount authorized for small businesses                        |
|--------------------------------------------|-------------------------------------------------------------------------|
| Woman Owned Authorized Amount              | Financing amount authorized for women-owned businesses                  |
|--------------------------------------------|-------------------------------------------------------------------------|
| Minority Owned Authorized Amount           | Financing amount authorized for minority-owned businesses               |
|--------------------------------------------|-------------------------------------------------------------------------|
| Loan Interest Rate                         | Interest rate applied to the loan                                       |
|--------------------------------------------|-------------------------------------------------------------------------|
| Multiyear Working Capital Extension        | Indicates if the deal is a multiyear working capital extension          |


**1.3.2 Data types**

In [4]:
# Data types
df.dtypes

Fiscal Year                                int64
Unique Identifier                         object
Deal Number                               object
Decision                                  object
Decision Date                             object
Effective Date                            object
Expiration Date                           object
Brokered                                  object
Deal Cancelled                            object
Country                                   object
Program                                   object
Policy Type                               object
Decision Authority                        object
Primary Export Product NAICS/SIC code     object
Product Description                       object
Term                                      object
Primary Applicant                         object
Primary Lender                            object
Primary Exporter                          object
Primary Exporter City                     object
Primary Exporter Sta

Most columns are stored as object (text), even for dates and numeric values. Preprocessing will be required to convert dates into datetime and ensure numeric fields (like ApprovedAmount, InterestRate) are properly typed for analysis.

**1.3.3 Sample Records (First 10 Rows)**

In [5]:
# First 10 rows
df.head(10)

Fiscal Year Unique Identifier Deal Number  Decision Decision Date  \
0         2007    08003048XA0002  AP003048AA  Approved    11/29/2006   
1         2007    08003048XA0003  AP003048AA  Approved    11/29/2006   
2         2007    08003048XA0004  AP003048AA  Approved     8/16/2007   
3         2007    08064737XM0001  AP064737XM  Approved     9/21/2007   
4         2007    08069888XG0003  AP069888XG  Approved    11/24/2006   
5         2007    08069888XG0004  AP069888XG  Approved      3/9/2007   
6         2007    08070062XH0002  AP070062XH  Approved     10/1/2006   
7         2007    08070062XI0001  AP070062XI  Approved     2/12/2007   
8         2007    08071060XJ0001  AP071060XJ  Approved     4/13/2007   
9         2007    08072367XI0001  AP072367XI  Approved     1/29/2007   

  Effective Date Expiration Date Brokered  Deal Cancelled  \
0            NaN      12/15/2011        No             No   
1     11/29/2006      12/15/2016        No             No   
2      8/16/2007       9/15/2017        No             No   
3      9/21/2007       7/31/2008        No             No   
4     11/24/2006        1/1/2007        No             No   
5       3/9/2007       4/30/2007        No             No   
6      10/1/2006       1/31/2007        No             No   
7      2/12/2007       1/31/2008        No             No   
8      4/13/2007        4/1/2008        No             No   
9      1/29/2007      11/30/2007        No             No   

                        Country          Program Policy Type  \
0  Private Export Funding Corp.        Guarantee         NaN   
1  Private Export Funding Corp.        Guarantee         NaN   
2  Private Export Funding Corp.        Guarantee         NaN   
3                 United States  Working Capital         NaN   
4                 United States  Working Capital         NaN   
5                 United States  Working Capital         NaN   
6                 United States  Working Capital         NaN   
7                 United States  Working Capital         NaN   
8                 United States  Working Capital         NaN   
9                 United States  Working Capital         NaN   

               Decision Authority Primary Export Product NAICS/SIC code  \
0                           Board                                   NaN   
1                           Board                                   NaN   
2                           Board                                   NaN   
3  Individual Delegated Authority                                  5088   
4  Individual Delegated Authority                                  3546   
5  Individual Delegated Authority                                  3546   
6  Individual Delegated Authority                                  3531   
7  Individual Delegated Authority                                  3531   
8  Individual Delegated Authority                                424910   
9  Individual Delegated Authority                                321999   

                                 Product Description        Term  \
0                                                NaN   Long Term   
1                                                NaN   Long Term   
2                                                NaN   Long Term   
3  Transportation Equipment and Supplies - Except...  Short Term   
4                             Power-Driven Handtools  Short Term   
5                             Power-Driven Handtools  Short Term   
6               Construction Machinery and Equipment  Short Term   
7               Construction Machinery and Equipment  Short Term   
8                 Farm Supplies Merchant Wholesalers  Short Term   
9  All Other Miscellaneous Wood Product Manufactu...  Short Term   

                         Primary Applicant  \
0       Private Export Funding Corporation   
1       Private Export Funding Corporation   
2       Private Export Funding Corporation   
3                     Bank Of America N.A.   
4              Ups Capital B

The first 10 rows show approvals from different years and programs. Some columns, like PolicyType and ProductDescription, have missing values, so cleaning is needed.

**1.3.4 Sample Records (Last 10 Rows)**

In [6]:
#  Last 10 rows
df.tail(10)

Fiscal Year Unique Identifier Deal Number  Decision Decision Date  \
51743         2025    09783646ST1001      783646  Approved     6/26/2025   
51744         2025    09783701ST1001      783701  Approved     6/24/2025   
51745         2025    09783706ST1001      783706  Approved     6/16/2025   
51746         2025    09783717ST1001      783717  Approved      6/9/2025   
51747         2025    09783748ST1001      783748  Approved     6/13/2025   
51748         2025    09783936ST1001      783936  Approved      6/2/2025   
51749         2025    09784129ST1001      784129  Approved      6/9/2025   
51750         2025    09784328ST1001      784328  Approved     6/11/2025   
51751         2025    09784488ST1001      784488  Approved     6/17/2025   
51752         2025    09784819ST1001      784819  Approved     6/25/2025   

      Effective Date Expiration Date Brokered  Deal Cancelled  \
51743            NaN             NaN       Yes             No   
51744       6/1/2025        6/1/2026       Yes             No   
51745       6/1/2025        6/1/2026       Yes             No   
51746       6/1/2025        6/1/2026       Yes             No   
51747       7/1/2025        7/1/2026       Yes             No   
51748       8/1/2025        8/1/2026       Yes             No   
51749      6/15/2025        7/1/2026       Yes             No   
51750       6/1/2025        6/1/2026       Yes             No   
51751            NaN             NaN       Yes             No   
51752            NaN             NaN        No             No   

                    Country    Program Policy Type  \
51743  Multiple - Countries  Insurance         ESC   
51744              Malaysia  Insurance         ESS   
51745             Indonesia  Insurance         ESS   
51746        United Kingdom  Insurance         ESS   
51747  Multiple - Countries  Insurance         ENB   
51748  Multiple - Countries  Insurance         EXP   
51749  Multiple - Countries  Insurance         EXP   
51750                Norway  Insurance         ESS   
51751                Mexico  Insurance        ESSU   
51752  Multiple - Countries  Insurance         EDS   

                   Decision Authority Primary Export Product NAICS/SIC code  \
51743  Individual Delegated Authority                                423860   
51744  Individual Delegated Authority                                424490   
51745  Individual Delegated Authority                                424490   
51746  Individual Delegated Authority                                311211   
51747  Individual Delegated Authority                                423390   
51748  Individual Delegated Authority                                326199   
51749  Individual Delegated Authority                                321920   
51750  Individual Delegated Authority                                423860   
51751  Individual Delegated Authority                                336999   
51752  Individual Delegated Authority                                325620   

                                     Product Description        Term  \
51743  Transportation Equipment and Supplies (except ...  Short Term   
51744  Other Grocery and Related Products Merchant Wh...  Short Term   
51745  Other Grocery and Related Products Merchant Wh...  Short Term   
51746                                      Flour Milling  Short Term   
51747   Other Construction Material Merchant Wholesalers  Short Term   
51748           All Other Plastics Product Manufacturing  Short Term   
51749            Wood Container and Pallet Manufacturing  Short Term   
51750  Transportation Equipment and Supplies (except ...  Short Term   
51751   All Other Transportation Equipment Manufacturing  Short Term   
51752                   Toilet Preparation Manufacturing  Short Term   

                         Primary Applicant          Primary Lender  \
51743                            DIMO Corp                     NaN   
51744         Nature's SunGrown Foods, LLC              

Reviewing the last 10 rows helps confirm consistency at the end of the dataset. The structure remains consistent, but categorical fields show missing or inconsistent entries.

**1.3.5 Info & Summary**

In [7]:
# Dataset info and summary statistics
df.info()
df.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51753 entries, 0 to 51752
Data columns (total 34 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   Fiscal Year                            51753 non-null  int64  
 1   Unique Identifier                      51753 non-null  object 
 2   Deal Number                            51746 non-null  object 
 3   Decision                               51753 non-null  object 
 4   Decision Date                          51753 non-null  object 
 5   Effective Date                         50512 non-null  object 
 6   Expiration Date                        51067 non-null  object 
 7   Brokered                               51753 non-null  object 
 8   Deal Cancelled                         51753 non-null  object 
 9   Country                                51753 non-null  object 
 10  Program                                51753 non-null  object 
 11  Po

Fiscal Year  Approved/Declined Amount  Disbursed/Shipped Amount  \
count     51753.00                  51753.00                  51753.00   
mean       2014.58                5020374.83                4282402.03   
std           4.90               42976783.08               32518553.98   
min        2007.00                      0.00                      0.00   
25%        2011.00                 200000.00                      0.00   
50%        2014.00                 500000.00                 300000.00   
75%        2018.00                1065607.83                1512522.74   
max        2025.00             3595031425.00             2942953633.00   

       Undisbursed Exposure Amount  Outstanding Exposure Amount  \
count                     51753.00                     51753.00   
mean                     251339.54                    344674.15   
std                    13221886.63                  12420356.41   
min                           0.00                         0.00   
25%                           0.00                         0.00   
50%                           0.00                         0.00   
75%                           0.00                         0.00   
max                  2046586551.00                2344488761.00   

       Small Business Authorized Amount  Woman Owned Authorized Amount  \
count                          51753.00                       51753.00   
mean                         1174483.00                       83516.70   
std                          6295936.61                      726896.55   
min                                0.00                           0.00   
25%                           100000.00                           0.00   
50%                           300000.00                           0.00   
75%                           800000.00                           0.00   
max                        333239610.50                    40000000.00   

       Minority Owned Authorized Amount  
count                          51753.00  
mean                          135462.26  
std                           957128.47  
min                                0.00  
25%                                0.00  
50%                                0.00  
75%                                0.00  
max                         36000000.00

**1.3.6 Info & Summary**

info() shows column counts, non-null values, and data types.

describe() provides statistical summaries for numerical features.  

Several columns have missing values (e.g., InterestRate, PolicyType). Numerical fields show skewness, with a few very large deals dominating totals. 

***1.4 Summary of Findings***

***Dataset Overview***

- Total Rows: 51,753 records
- Total Columns: 34 features (before cleaning)
  
***Data Profile***
  
**Fiscal Year:** Records span FY25 Q3 (Oct 2024 – Sep 2025)

**Approved Amounts:** Range from small loans (<1M dollar) to very large deals (> 100M dollar)

**Programs:** Includes Working Capital, Guarantees, and other financing types

**Business Support:** Columns tracks financing for small, women‑owned, and minority‑owned enterprises

***Data Quality Issues Identified***

**Missing Values:**

- InterestRate: ~51,000 missing (almost entire column)

- PolicyType: ~9,000 missing

- ProductDescription: ~1,500 missing

- ExporterState: ~1,600 missing

**Duplicate / Redundant Columns:**

- Identifiers (Unique Identifier, Deal Number) add little analytical value

**Data Type Issues:**

- Dates stored as text (DecisionDate, EffectiveDate, ExpirationDate)

- Numeric fields like ApprovedAmount sometimes mixed with text formatting

**Inconsistent Values:**

- Categorical fields (e.g., PolicyType, ExporterState) contain blanks or “Unknown”

***Key Business Insights***

- Approvals are concentrated in a few countries and program types, showing dependency risks.

- Financing amounts are highly skewed — most deals are small, but a few very large ones dominate totals.

- Support for small, women‑owned, and minority‑owned businesses exists but is limited compared to overall financing.

- Outstanding exposures highlight areas of financial risk, requiring closer monitoring.